### DISCUSSION TITLE
Grace notes on a different string

### DISCUSSION THREAD

**Discussion by Nastase** [helge17/tuxguitar#338] [General]
Hi,

I'm rather new to TuxGuitar and I was trying to write something like this:
![image](https://github.com/helge17/tuxguitar/assets/815742/739c8a19-607a-4310-8b00-381aa8bcdff6)

So basically, what I'm trying to do is to have a grace note before the beat but on a different string. So far what I've managed to do in TuxGuitar is to have a grace note on a different fret but on the same string as the "related" note.

Is what I'm trying to achieve possible?

Thank you!

**Comment by guiv42** [comment]
Hi.
No, currently this is **not** possible. If you would like this feature, I suggest creating a new feature request (see [issues](https://github.com/helge17/tuxguitar/issues)).
That might be a little tricky to implement, especially when a note is moved from one string to another: where should such a grace note go?

By the way, I checked the implementation, and currently this case is **not** correctly handled. I've just created #339 to track this bug.

**Reply by Nastase** [reply]
Hi, thanks for your reply @guiv42, I'll create the feature request.

Meantime, I got over the bug you opened. I'm not knowledgeable on the implementation code-wise, but as someone looking from the "outside", couldn't grace notes be considered independent notes?

So one would create a note on any string and just mark it as a grace note, instead of adding a grace note to an already created note. Basically, that would be marking a note as one that doesn't influence the time counting on the measure. And then one could apply all the "normal" embellishments (hammer, bend, etc...) on the grace note, just as a normal note.

I know grace notes are transitions to the "real" note we want to play, but without being tied to another note I reckon it would be easier implementation wise, specially in all the cases you mention in the bug when moving notes around.

Hope this can help.

**Reply by guiv42** [reply]
It seems to me we first need to fix the existing behavior. Then we can see how to implement such an evolution. As other pending feature requests, whatever the implementation it requires to change the data structure of .tg files. So that will require some time.
In terms of data structure, I think the easiest is to keep the association of grace note to another "real" note. What would an "isolated" grace note mean? And any action on the "real" note shall have also a consequence of the associated "grace": e.g. delete, copy/paste, move, etc.
